oauth2client/contrib/gce.py

```diff
@@ -65,24 +61,41 @@ def __init__(self, scope='', **kwargs):
                    the actual scopes for tokens requested. These scopes are
                    set at VM instance creation time and won't change.
         """
-        if scope:
+        if 'scopes' in kwargs:
             warnings.warn(_SCOPES_WARNING)
-        # This is just provided for backwards compatibility, but is not
-        # used by this class.
-        self.scope = util.scopes_to_string(scope)
-        self.kwargs = kwargs
+            kwargs['scopes'] = None
 
         # Assertion type is no longer used, but still in the
         # parent class signature.
-        super(AppAssertionCredentials, self).__init__(None)
+        super(AppAssertionCredentials, self).__init__(None, *args, **kwargs)
 
-        # Cache until Metadata Server supports Cache-Control Header
-        self._service_account_email = None
+        self.service_account_email = email
+        self.scopes = None
+        self.invalid = True
 
     @classmethod
     def from_json(cls, json_data):
-        data = json.loads(_from_bytes(json_data))
-        return AppAssertionCredentials(data['scope'])
+        raise NotImplementedError(
+            'Cannot serialize credentials for GCE service accounts.')
+
+    def to_json(self):
+        raise NotImplementedError(
+            'Cannot serialize credentials for GCE service accounts.')
+
+    def _retrieve_scopes(self, http_request):
```

Comment:
`_retrieve_scopes` is an override of `Credentials._retrieve_scopes` which is called by `Credentials.retrieve_scopes` which references `httplib2` specifically, so this seemed like the best place to hook in.